co-simulation/fed/mosaic-carma-messenger/src/main/java/org/eclipse/mosaic/fed/carmamessenger/ambassador/CarmaMessengerInstance.java

```diff
@@ -0,0 +1,142 @@
+    private DatagramSocket rxMsgsSocket = null;
+
+    private InetAddress targetAddress;
+    private int v2xPort;
+    private int timeSyncPort;
+    private GeoPoint location = GeoPoint.ORIGO;
+    private String messengerEmergencyState;
+
+    public CarmaMessengerInstance(String carmaVehicleId, String carlaRoleName, InetAddress targetAddress, int v2xPort, int timeSyncPort, String messengerEmergencyState) {
+        this.carmaVehicleId = carmaVehicleId;
+        this.carlaRoleName = carlaRoleName;
+        this.targetAddress = targetAddress;
+        this.v2xPort = v2xPort;
+        this.timeSyncPort = timeSyncPort;
+        this.messengerEmergencyState = messengerEmergencyState;
+    }
+
+    public InetAddress getTargetAddress() {
+        return targetAddress;
+    }
+
+    public void setTargetAddress(InetAddress targetAddress) {
+        this.targetAddress = targetAddress;
+    }
+
+    public void setLocation(GeoPoint location) {
+        this.location = location;
+    }
+
+    public GeoPoint getLocation() {
+        return this.location;
+    }
+    
+
+    public int getV2xPort() {
+        return v2xPort;
+    }
+
+    public void setV2xPort(int v2xPort) {
+        this.v2xPort = v2xPort;
+    }
+
+    public int getTimeSyncPort() {
+        return timeSyncPort;
+    }
+
+    public void setTimeSyncPort(int timeSyncPort) {
+        this.timeSyncPort = timeSyncPort;
+    }
+
+    /**
+     * Sends the V2X message to the CARMA Platform communications interface configured at construction time.
+     * @param data The binary data to transmit
+     * @throws IOException If there is an issue with the underlying socket object or methods
+     */
+    public void sendV2xMsgs(byte[] data) throws IOException {
+        if (rxMsgsSocket == null) {
+            throw new IllegalStateException("Attempted to send data before opening socket");
+        }
+
+        DatagramPacket packet = new DatagramPacket(data, data.length, targetAddress, v2xPort);
+
+        rxMsgsSocket.send(packet);
+    }
+    /**
+     * Sends the time sync messages to the CARMA Platform to synchronize ros clock with simulation clock.
+     * @param data The binary data encoding of json time sync message
+     * @throws IOException If there is an issue with the underlying socket object or methods
+     */
+    public void sendTimeSyncMsg(byte[] data) throws IOException {
+        if (rxMsgsSocket == null) {
+            throw new IllegalStateException("Attempted to send data before opening socket");
+        }
+
+        DatagramPacket packet = new DatagramPacket(data, data.length, targetAddress, timeSyncPort);
+
+        rxMsgsSocket.send(packet);
+    }
+
+    /**
+     * Connects the sockt to receive messages from the CARMA Platform instance
+     * @throws IOException If the socket creation fails
+     */
+    public void bind() throws IOException {
+        rxMsgsSocket = new DatagramSocket();
+    }
+
+    public String getCarmaVehicleId() {
```

Comment:
Do we still use the function name same with CARMA Ambassador or we can change to something like "carmaMessengerVehicleId"?